File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/model/MasterModule.java
package com.cdac.sarvam.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonManagedReference;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "master_modules")
public class MasterModule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer moduleId;

    private String moduleName;
    private LocalDateTime moduleStartDate;
    private LocalDateTime moduleEndDate;

    @ManyToOne
    @JoinColumn(name = "instructor_id", nullable = false)
    private MasterProfile instructor;

    @ManyToOne
    @JoinColumn(name = "moco_id", nullable = false)
    private MasterProfile moCo;

    private Byte noOfDays;
    private Byte durationHours;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    @OneToMany(mappedBy = "module", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<MasterResult> masterResults;
    
    @OneToMany(mappedBy = "module", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<MasterModuleSubpoint> masterModuleSubpoints;
    
    @OneToMany(mappedBy = "module")
    private List<Question> questions;
}

File: MS.NET/Sarvam Backend/Models/MasterModule.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarvam.Models
{
    public class MasterModule
    {
        [Key]
        public int ModuleId { get; set; } // Primary Key (PK)
        public string ModuleName { get; set; }
        public DateTime ModuleStartDate { get; set; }
        public DateTime ModuleEndDate { get; set; }
        public long InstructorId { get; set; } // FK to MasterProfile.PRN (Instructor)
        public long MoCoId { get; set; } // FK to MasterProfile.PRN (MoCo)
        public byte NoOfDays { get; set; }
        public byte DurationHours { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;

        // Navigation Properties
        [NotMapped]
        public MasterProfile Instructor { get; set; } // Navigation Property for Instructor (MasterProfile)
        [NotMapped]
        public MasterProfile MoCo { get; set; } // Navigation Property for MoCo (MasterProfile)

        // Related subpoints for the module
        [NotMapped]
        public ICollection<MasterModuleSubpoint> MasterModuleSubpoints { get; set; }

        // One-to-Many Relationship with Questions
        public ICollection<Question> Questions { get; set; }

        // One-to-Many Relationship with MasterResults
        public ICollection<MasterResult> MasterResults { get; set; }

    }
}
